### DISCUSSION TITLE
How to get distance between two nodes in a plugin.

### DISCUSSION THREAD

**Discussion by Dr-Goodbeard** [TUDelft-CNS-ATM/bluesky#466] [Q&A] [answered]
Hi, I'm wanting to use the built in distance function "dist" in my plugin to find the distance between two nodes (aircraft to waypoint or waypoint to waypoint). 

I can do this with stack.stack(dist wpt1 wpt2) but how do I use the output of that in my plugin?

Thanks,

**Comment by jooste** [comment]
Hi @nl22-nmsu,

The stack functions only give their output to the bluesky console. If you want to use the results in your plugin you should directly call the underlying functions, which reside in `bluesky.tools.geo`. Have a look at the [wiki](https://github.com/TUDelft-CNS-ATM/bluesky/wiki/geo) for more info.

**Reply by Dr-Goodbeard** [reply]
So I am using that function for the distance calculation. Right now I'm struggling on getting the locations for certain airports. 

I can get the location of the aircraft using traf.lat, but is there a similar thing for the airports? For example, I want the location for "ABQ".

**Reply by jooste** [reply, marked as answer]
You can find a description of the list of attributes of the navigation database on the [wiki](https://github.com/TUDelft-CNS-ATM/bluesky/wiki/navdb). All of these are Python lists with the same order. The idea is therefore that you find the index of a named waypoint from `navdb.wpid`, and then use that index to access waypoint property lists like wplat/wplon. The navdb also has a function `getwpindices(self, wpid, reflat, reflon,crit)` with which you can find the most appropriate waypoint for a given id (many waypoint names are used multiple times across the globe).

**Reply by Dr-Goodbeard** [reply]
Oh perfect, that was super helpful, thank you!
